api/src/test/java/org/openmrs/module/locationbasedaccess/aop/EncounterSearchAdviserTest.java

```diff
@@ -0,0 +1,861 @@
+        assertNull(encounter5);
+        
+        Encounter encounter6 = encounterService.getEncounterByUuid("y403fafb-e5e4-42d0-9d11-4f52e89d123r");
+        assertNull(encounter6);
+        
+        Encounter encounter7 = encounterService.getEncounterByUuid("9519d653-393b-4118-9c83-a3715b82d4ac");
+        assertNull(encounter7);
+        
+        Encounter encounter8 = encounterService.getEncounterByUuid("0619d653-393b-4118-9c83-a3715b82d4ac");
+        assertNull(encounter8);
+    }
+    
+    @Test
+    public void getEncounters_adminCanAccessEncounterFromAllLocations() {
+        Context.authenticate("admin", "test");
+        User authenticatedUser = Context.getAuthenticatedUser();
+        assertNotNull(authenticatedUser);
+        assertTrue(authenticatedUser.isSuperUser());
+        
+        Location location1 = locationService.getLocation(DEMO_LOCATION1_ID);
+        Location location2 = locationService.getLocation(DEMO_LOCATION2_ID);
+        
+        Patient patient1 = patientService.getPatient(2);
+        EncounterSearchCriteria esc1 = new EncounterSearchCriteria(patient1,null,null,null,null,null,null,null,null,null,false);
+        List<Encounter> encounterList1 = encounterService.getEncounters(esc1);
+        assertEquals(encounterList1.size(),5);
+        Encounter encounter1 = encounterList1.get(0);
+        Encounter encounter2 = encounterList1.get(1);
+        Encounter encounter3 = encounterList1.get(2);
+        Encounter encounter4 = encounterList1.get(3);
+        Encounter encounter5 = encounterList1.get(4);
+        assertNotNull(encounter1);
+        assertEquals(encounter1.getLocation(),location1);
+        assertNotNull(encounter2);
+        assertEquals(encounter2.getLocation(),location2);
+        assertNotNull(encounter3);
+        assertEquals(encounter3.getLocation(),location2);
+        assertNotNull(encounter4);
+        assertEquals(encounter4.getLocation(),location2);
+        assertNotNull(encounter5);
+        assertEquals(encounter5.getLocation(),location2);
+        
+        Patient patient2 = patientService.getPatient(7);
+        EncounterSearchCriteria esc2 = new EncounterSearchCriteria(patient2,null,null,null,null,null,null,null,null,null,false);
+        List<Encounter> encounterList2 = encounterService.getEncounters(esc2);
+        assertEquals(encounterList2.size(),3);
+        Encounter encounter6 = encounterList2.get(0);
+        Encounter encounter7 = encounterList2.get(1);
+        Encounter encounter8 = encounterList2.get(2);
+        assertNotNull(encounter6);
+        assertEquals(encounter6.getLocation(),location1);
+        assertNotNull(encounter7);
+        assertEquals(encounter7.getLocation(),location1);
+        assertNotNull(encounter8);
+        assertEquals(encounter8.getLocation(),location2);
+    }
+    
+    @Test
+    public void getEncounters_nonAdminUserWithOutLocationPropertyShouldNotGetEncounterInformation(){

```

Comment:
Have you checked for null user property?

Reply:
Fixed 😃